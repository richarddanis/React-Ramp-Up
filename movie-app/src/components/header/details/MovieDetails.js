import React from "react";
import NetflixRouletteName from "../../util/NetflixRouletteName";
import Card from "../../util/Card";
import {useDispatch} from 'react-redux';
import * as actionType from '../../../store/actions/actions'

function MovieDetail({movie}){
    const dispatch = useDispatch();

    function handleCloseMovieDetails(){
        dispatch(actionType.handleCloseDetailsSection());
    }

    return(
        <div className="uk-container-expand uk-panel" >
            <Card closeEvent={handleCloseMovieDetails}>
                 <div className="uk-grid-small uk-child-width-expand@s" data-uk-grid>
                    <NetflixRouletteName/>
                </div>
                    <div className="uk-grid-small uk-child-width-expand@s" data-uk-grid>
                        <div className="uk-card-media-left uk-cover-container">
                            <img src={`${movie.poster_path}`} alt="" data-uk-cover/>
                            <canvas width="" height=""></canvas>
                        </div>
                        <div className="uk-card-body" style={{color: 'white'}}>
                            <h3 className="uk-card-title"><span className="color-white">{[`${movie.title}`,`${movie.vote_average}`].join(' ')}</span></h3>
                            <p>{movie.genres.map(genre => genre).join(' & ')}</p>
                            <span style={{color: '#F65261'}}>{movie.release_date}</span>
                            <p>{movie.overview}</p>
                        </div>
                    </div>
            </Card>
        </div>
    );
}

export default MovieDetail;