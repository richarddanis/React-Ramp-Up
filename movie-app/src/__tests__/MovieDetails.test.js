import { shallow } from 'enzyme';
import React from 'react';
import MovieDetails from '../components/header/Header';

test('shallow renders app component', () => {
    shallow(<MovieDetails />);
});
