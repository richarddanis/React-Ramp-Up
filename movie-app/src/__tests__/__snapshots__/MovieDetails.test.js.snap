// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModals shallow renders app component 1`] = `
<div
  className="uk-position-fixed uk-position-top uk-position-left center-card-position"
  style={
    Object {
      "backgroundColor": "#0008",
    }
  }
>
  <div
    className="uk-position-center"
  >
    <div>
      <div>
        <div
          className="uk-card-default uk-card-body uk-border-rounded"
          style={
            Object {
              "backgroundColor": "#232323",
            }
          }
        >
          <div
            className="uk-card-badge"
          >
            <button
              className="uk-button background-transparent"
              onClick={[Function]}
            >
              X
            </button>
          </div>
          <div
            className="uk-margin"
          >
            <div>
              <h1
                className="color-white"
              >
                Delete movie
              </h1>
              <p>
                Are you sure want to delete the movie?
              </p>
              <button
                className="uk-button background-transparent"
                onClick={[Function]}
              >
                Confirm
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
